File: completable-reactor-graph-viewer/src/main/java/ru/fix/cpapsm/completable/reactor/graph/viewer/CoordinateTranslator.java
package ru.fix.cpapsm.completable.reactor.graph.viewer;

import lombok.NonNull;
import lombok.RequiredArgsConstructor;

/**
 * Created by Kamil Asfandiyarov
 */
@RequiredArgsConstructor
public class CoordinateTranslator {

    @NonNull
    Double worldSize;

    Double translateX(Integer x) {
        return worldSize / 2 + x;
    }

    Double translateX(Double x) {
        return worldSize / 2 + x;
    }

    Double translateY(Integer y) {
        return worldSize / 2 + y;
    }

    Double translateY(Double y) {
        return worldSize / 2 + y;
    }

    int reverseTranslateX(Double x){
        return (int)(x - worldSize / 2);
    }

    int reverseTranslateY(Double y){
        return (int)(y - worldSize / 2);
    }
}

File: completable-reactor-graph-viewer/src/main/kotlin/ru/fix/completable/reactor/graph/viewer/gl/CoordinateTranslator.kt
package ru.fix.completable.reactor.graph.viewer.gl

/**
 * Created by Kamil Asfandiyarov
 */

class CoordinateTranslator (val graphPane: GraphPane) {

    fun paneLayoutToModelX(x: Double): Int {
        return (graphPane.graphBordersInModelCoordinates.minX.toDouble() + x).toInt()
    }


    fun paneLayoutToModelY(y: Double): Int {
        return (graphPane.graphBordersInModelCoordinates.minY.toDouble() + y).toInt()
    }
}